c++/src/capnp/serialize-async.h

```diff
@@ -101,6 +91,29 @@ class MessageTransport {
       kj::ArrayPtr<const int> fds,
       kj::ArrayPtr<const kj::ArrayPtr<const word>> segments) = 0;
     KJ_WARN_UNUSED_RESULT;
+  // The tryReadMessage & writeMessages methods have the same contracts as
+  // the corresponding FD-oriented stand-alone functions above, except that the
+  // first argument (of type AsyncCapabilityStream) is omitted.
+  //
+  // Below, we provide helper functions that correspond to the rest of the top-level
+  // functions above, but take a MessageTransport& as their first argument.
+  //
+  // Implementations which do not support FD passing may simply ignore the relevant
+  // arguments.
+
+  virtual kj::Maybe<int> getSendBufferSize() = 0;
+  // Get the size of the underlying send buffer, if applicable. The RPC
+  // system uses this as a hint for flow control purposes; see:
+  //
+  // https://capnproto.org/news/#multi-stream-flow-control
+  //
+  // ...for a more thorough explanation of how this is used. Implementations
+  // may return nullptr if they do not have access to this information, or if
+  // the underlying transport does not use a congestion window.
+
+  virtual kj::Promise<void> shutdownWrite() = 0;
```

Comment:
Let's call this `end()`. I'm working on a refactor of `AsyncOutputStream` which replaces `shutdownWrite()` with `end()` (returning a promise, like you do here).